### ISSUE TITLE
Separate dataDirHostPath from image version logic in external cluster validation

### ISSUE THREAD

**Issue by briantopping** [rook/rook#7305]
**Is this a bug report or feature request?**
* Feature request

**What should the feature do:** 
An external cluster configured with an image version will be rejected by [this code](https://github.com/rook/rook/blob/8b753e2923819dea8da6643e9355c5c461dbc685/pkg/operator/ceph/cluster/cluster_external.go#L180-L184) if `dataDirHostPath` is not defined. It does not matter if the features that require `dataDirHostPath` (such as CephFS or RGW) are being used or not. This creates an unusable situation when the image version is required in hyperscaler environments.

**What is use case behind this feature:**
In hyperscalers such as EKS/GKE/etc, the control plane runs in an inaccessible administrative domain of the cluster and the worker nodes are ephemeral. Allocating storage on host nodes is generally dangerous in such a situation. Over time, the control plane is likely to destroy nodes that host this data with unclear outcomes. For instance, does the external cluster retain integrity if only RBD is used and the host storage is lost?

In cases where a specific Rook image is not required, the current code poses no problems, correctly reducing CR options. But issues such as #6849 mean that an image may be specified for the external cluster to match an upstream cluster version that is held back and the validation logic should not block. There may be user temptation to specify `dataDirHostPath`, but that may simply lead downstream users to create CRs that not backed by reliable storage. 

To resolve this, `dataDirHostPath` should not be a feature gate. Specify the feature gates separately and determine validity of a missing or empty `dataDirHostPath` by the use of the feature gates.

**Comment by leseb** [comment]
I think I'm confused. The only reason why you would pass an image version is if you are looking at deploying RGW/MDS on the control plane which does not seem to be your use case.
The issue you pointed reflects an OSD deployment issue, OSDs are **only** deployed and managed in converged mode, so again it looks like your use case is safe.

Am I missing something?

**Comment by briantopping** [comment]
Thanks @leseb, I came to this use case initially because my external cluster wasn't loading, after looking at the code, I saw that it was because I was specifying a container version to get around #6849. My external clients don't need RGW and CephFS, but I believe the features are unlocked if `dataDirHostPath` is defined (this issue should be closed if that's not the case). It's unclear if some _other_ user would inadvertently use RGW or CephFS if they didn't know it was dangerous because the data resided on host volumes of ephemeral nodes.

Put another way, I'm trying to lock out RGW and CephFS when `dataDirHostPath` is defined on an ephemeral volume.

So it's not that my use case for RBD doesn't work, but it seems like there should be a better interlock against `dataDirHostPath` being defined simply to allow an image to be defined and inadvertently unlocking features that may allow data loss due to the underlying configuration.
